### ISSUE TITLE
Could Finger tool select areas too?

### ISSUE THREAD

**Issue by F-Burning** [opentoonz/opentoonz#3839]
I know this topic was formulated for @konero years ago #2539, but it was closed, and I was wondering if this tool nowadays could select areas automatically too, to expand colors easy.

![Finger tool + areas](https://user-images.githubusercontent.com/74098734/113127781-10452b80-9211-11eb-8fbf-6c782c169210.jpg)

This, although its initial purpose was another, is very similar to Paintman's finger tool, and helps a lot to fill holes easily.

I know exists the rectangular fill tool, but with the finger you only need to do one step, while with fill you have to select colors one by one with the style picker and select again the fill tool.

Thank you for reading this far.

**Comment by F-Burning** [comment]
> You can skip using the style picker with the fill tool. Instead change styles by using number hotkeys. Ie. numbers 0 - 9, will pick the first 10 styles (where 1 = the first style & so on). If you have more than 10 styles, use the tab key to switch between sets of 10.

In my opinion, that's a lot of thinks that you have to be concentrated to only do an operation that could be solved with one click 😅 
 
That is my dissatisfaction with Opentoonz, because it's very powerful, but while in other softwares you solve a problem with a click, here you have to take many more unnecessary steps to do the same.

I love the software and I really appreciate the almost daily work that is done here, but at least I would like to highlight a few things that I think could go for the better, in my humble opinion.

**Comment by KaseyFarron** [comment]
Agree! Numpad is nice for initial areas but at the end when you spot some unfilled gaps it can be time consuming to use.

I made a script in AutoHotkey that auto picks the style under the cursor, switches to freehand fill tool mode and makes a selection all in one click & drag, it was annoying me as well needing to select styles manually each time.

To activate it, I need to hold F9 key down, it will then run when Left Mouse Button is pressed. I had to set shortcuts for **Fill Tool - Areas** (shift+F) and **Style Picker - Areas** (shift+K) in OpenToonz.

Here's an example, with F9 pressed down I just need to click and drag once, I can keep F9 held and make lots of mouse drags:

![fill_unfilled_gaps](https://user-images.githubusercontent.com/55353161/113289941-d34c6800-92e8-11eb-8d3b-24868863368b.gif)

I know it's not the same as color drag perhaps, but I thought this might be somehow easier to implement by adding some option to auto pick style at first click point when a special key is held down. Would be nice if it was built in! 😹 

If you'd like to try it, I [uploaded it on GitHub](https://github.com/KaseyFarron/ColourPick-FreehandFill).

**Comment by artisteacher** [comment]
That's fantastic!

**Comment by beeheemooth** [comment]
@KaseyFarron Perfect solution! I also planned to make a similar macro, but you got ahead of me. It's a good idea to host macros here on github!
I think that this behavior of the Fill Tool should be by default (when the Freehand, Areas options are active), for example, when the Alt button is held down.

**Comment by F-Burning** [comment]
Really good @KaseyFarron! 😄

Could you tell me how to install it?

Thank you!!

> I think that this behavior of the Fill Tool should be by default (when the Freehand, Areas options are active), for example, when the Alt button is held down.

This!

Even with being able to put the Alt key to the style selector in the keyboard shortcuts it would already be an advance 😁

I hope this thread does not fall into oblivion and is taken into consideration 🙏

**Comment by KaseyFarron** [comment]
Thank you for the kind comments!

@F-Burning I [updated my GitHub page](https://github.com/KaseyFarron/AutoPickColourAndFill) with more detailed instructions I hope is more clear for how to use and setup!

I also added CLIP Studio Paint support and added an external `settings.ini` file so the keys can be customized if necessary. 😋 

But I agree, hopefully OT will do this natively in the future! 🤩

**Comment by F-Burning** [comment]
@KaseyFarron thank you soo much! 😀 

I don't know why, but this not work for me.

I'm using the portable.exe and then follow the instructions, but nothing happens.

Anyway, it's a wonderful feature, and I hope this will be in a future too 😄

**Comment by oliver-dreamwalker** [comment]
This feature request should get more attention.

**Comment by Abel032** [comment]
I think it won't be hard to implement.
Add two opintion:
1. whether to pick style (off as default)
2. fill Area or fill Ink

**Comment by RodneyBaker** [comment]
This has been implemented and is available in the current nightly release.
